lib/guard/yard/server.rb

```diff
@@ -58,6 +50,30 @@ def verify
         )
         false
       end
+
+      private
+
+      def windows_kill
+        # /t includes child process
+        # /f forces (required else we'll fail)
+        if system("taskkill /f /t /pid #{pid}").nil?
```

Comment:
It seems strange that this is the way a process must be killed on Windows. Does Ruby not provide a standard library implementation here?

Reply:
This plugin ends up spawning a process tree (3 processes). You can kill some of them using Ruby's patched in behavior for signal handling but not all. 

There's the Win32 Way to do things. 

stdlib has [this](http://ruby-doc.org/stdlib-2.4.2/libdoc/win32ole/rdoc/index.html). But it does not support processes. For that one would need to require [one of these](https://rubygems.org/search?utf8=%E2%9C%93&query=win32).

Reply:
Okay. I am okay with your proposed implementation. It might be preferable to use another gem in the long term, but I cannot test the implementation on windows and I am not knowledgeable enough about releasing a gem with dependencies that only apply to one OS to make that change. If this is something you feel comfortable tackling, I would be grateful and happy to take a look.

One comment on this line. I would prefer if the conditional is simplified to account for `nil` and `false` return values:

```
unless system(...)
```

Reply:
You prefer using a Gem for Windows to do this? 

If Gem, the way to go about it is to *not* add it to the gemspec (since you cannot specify platform specific dependencies) and to just require a runtime. You can add a warning to the gemspec via `post_install_message`. 

But really not sure what the benefit is. Pretty sure `taskkill` is available on on versions on Windows since XP

> I would prefer if the conditional is simplified to account for nil and false return values:
> unless system(...)


`false` means process did not exist. `nil` means command was not found. Based on current POSIX version, if the process did not exist we want to continue without error.

Reply:
Okay. I see I misunderstood the `system` return values. Your implementation LGTM then.

I would prefer if we could have platform dependent gems in the gemspec so that we could depend on another Gem to do this work 😀 but that's not possible, so I think your solution is the optimal for the conditions.